components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/Utils.kt

```diff
@@ -73,6 +73,17 @@ suspend fun handleFxaExceptions(logger: Logger, operation: String, block: () ->
     })
 }
 
+/**
+ * Helper method that intercepts [FxaException]s thrown when executing [block].
+ * - Adds own logic for handling such exceptions
+ * - Retrows the unaltered exceptions to callers of this method
+ */
+suspend fun executeAndGetFxaExceptions(logger: Logger, operation: String, block: () -> Unit) {
```

Comment:
If your plan is to just re-throw whatever it is `block` throws, you don't need this wrapper at all. Just execute `block` directly!